pkg/fqdn/dnsproxy/proxy_test.go

```diff
@@ -1000,6 +1002,35 @@ func (s *DNSProxyTestSuite) TestRestoredEndpoint(c *C) {
 	_, exists = s.proxy.restored[epID1]
 	c.Assert(exists, Equals, false)
 
+	invalidRePattern := "invalid-re-pattern((*"
+	validRePattern := "^this[.]domain[.]com[.]$"
+	k := maps.Keys(restored)[0]
```

Comment:
The order doesn't matter since this map only has one element.

I _think_ the key is actually random since its the port of the dns-server, and it listens to `:0` that I think means "any free" port or so. Guess we could parse `s.proxy.TCPServer.Listener.Addr()` as well, but I thought this was the simplest. Maybe I should add a comment about it

ref. https://github.com/cilium/cilium/blob/e28cce148c676371a234a51f303c8c7933f81960/pkg/fqdn/dnsproxy/proxy_test.go#L120